src/typechecker.erl

```diff
@@ -4426,6 +4442,19 @@ is_list_pat_exhaustive({nil, _}) -> true;
 is_list_pat_exhaustive({cons, _, {var, _, _}, {var, _, _}}) -> true;
 is_list_pat_exhaustive(_) -> false.
 
+%% TODO: This is incomplete!
+%% To properly check pattern exhaustiveness we have to consider bound variables.
+%% Pattern: <<>>
+is_bin_pat_exhaustive([]) ->
+    true;
+%% Pattern: <<_, _/bytes>>
+is_bin_pat_exhaustive([{bin_element, _, {var, _, _}, _, _},
```

Comment:
This function doesn't check if a pattern is exhaustive (what the name indicates) but rather if the pattern can be used for exhaustiveness checking or if exhaustiveness checking should be disabled.

I think we should come up with a better name for this function. (We can negate the return values or change them if it fits the function name better.)

Reply:
Yes, you're right. I was trying to avoid too long and complicated names, but it seems unavoidable if we want to capture the meaning correctly.